ir-engine/ir-engine
# IR-4861 avatar nameplates

## Summary
bringing back avatar nameplates, distance based activation, only shows on other player avatars (not your own), displays username, transitions in and out with scale, height and opacity. Camera frustum culling also being used

## References
ticket:
[https://tsu.atlassian.net/browse/IR-4861](https://tsu.atlassian.net/browse/IR-4861)

design:
[https://www.figma.com/design/xG10l6SZBn92pFmPVMWfBG/Viewer-%26-XRUI?node-id=30-6852&t=jm0FDfyqsahXkwjp-1](https://www.figma.com/design/xG10l6SZBn92pFmPVMWfBG/Viewer-%26-XRUI?node-id=30-6852&t=jm0FDfyqsahXkwjp-1)

## QA Steps
join a location with two players (or one normal, one incognito tab) and view avatar nameplate above other player. 
you should see:
- distance threshold is roughly 8 meters
- scaling in/out when within distance threshold
- nameplate rises/lowers in/out
- nameplate fades in/out
- transition time of 0.25 seconds
- basic camera frustum culling (if the center of the nameplate is outside the camera view it will close
